plugins/hfp_ag_bluez5.c

```diff
@@ -344,7 +407,9 @@ static int hfp_ag_init(void)
 	__ofono_modem_foreach(call_modemwatch, NULL);
 
 	connection_hash = g_hash_table_new_full(g_str_hash, g_str_equal,
-					g_free, connection_destroy);
+					free_device_info, connection_destroy);
```

Comment:
The third argument to g_hash_table_new_full() is for destroying the key, so g_free is still needed there. You need to modify connection_destroy() instead of defining free_device_info() and free the memory for the structure there.